Added Helm unit tests for CoreDNS chart components

vinaycharlie01 opened the pull request:
#### Why is this pull request needed and what does it do?
  - Added unit tests for ClusterRole, ClusterRoleBinding, ConfigMap, Deployment, HPA, PDB, Service, ServiceMonitor, and ServiceAccount
  - Included autoscaler-specific test files
  - Updated GitHub Actions workflow to run chart testing
  
#### Which issues (if any) are related?
fix #208 


Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/coredns/helm/blob/master/CONTRIBUTING.md#versioning).
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/coredns/helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/coredns/helm/blob/master/CONTRIBUTING.md#developer-certificate-of-origin).

Changes are automatically published when merged to `main`. They are not published on branches.

<details>
  <summary>Note on DCO</summary>

  If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

hagaibarel commented:
I really like this direction, but there are a couple of things we should do to get this merged:
1. Please rebase your PR on top the latest changes and bump the chart version. It should be a minor bump not a patch one. Also make sure the artifact annoations are correct
2. Add the `tests` folder to `.helmignore`, they shouldn't be part of the published chart
3. We should probably add a reference in the `README` about writing unit-tests and debugging locally

Is there a way to get coverage report from the plugin? That way we can validate that changes are covered by unit tests

Also, note that the DCO check is failing, there are probably some commits which are not signed properly

shubham-cmyk commented:
https://github.com/helm-unittest/helm-unittest/blob/main/DOCUMENT.md
This particular doc explains how to write these BDD test.

Also @vinaycharlie01 can we put 

```
# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
```
on top of test files.

vinaycharlie01 commented:
> I really like this direction, but there are a couple of things we should do to get this merged:
> 
> 1. Please rebase your PR on top the latest changes and bump the chart version. It should be a minor bump not a patch one. Also make sure the artifact annoations are correct
> 2. Add the `tests` folder to `.helmignore`, they shouldn't be part of the published chart
> 3. We should probably add a reference in the `README` about writing unit-tests and debugging locally
> 
> Is there a way to get coverage report from the plugin? That way we can validate that changes are covered by unit tests
> 
> Also, note that the DCO check is failing, there are probably some commits which are not signed properly

Thanks for the detailed feedback!

I'll go ahead and:

- Rebase the PR on the latest changes.
- Bump the chart version with a minor update.
- Add the tests/ folder to .helmignore.
- Update the README with instructions on writing and debugging unit tests
Currently, Helm unittest doesn't support coverage reports, but I’ve added JUnit test result publishing.
Please let me know if this is required—I’ve included a screenshot below for reference.

<img width="1728" alt="image" src="https://github.com/user-attachments/assets/e698461f-2a94-4c70-9723-ccd3cc57ad96" />

I’ve added unit tests in the pipeline as well:

<img width="1728" alt="image" src="https://github.com/user-attachments/assets/bd4e9ab9-dbe5-4e0e-be0d-6725f62659f5" />

Appreciate the review I'll push the updates shortly!
